### PR TITLE
[Refactor] Moved duplicate code into `SetRealDomainAndAddress`

### PR DESCRIPTION
**Pull request by scowalt**
`GetRasterizedUnits` has some duplicate logic for setting memory domains. This change refactors that duplicate logic into a helper function.

Since one chunk uses `SourceDomain` and another `Domain`, there wasn't a super clean way for me to refactor this. I still think this deduplication is a net positive.

### CODE REVIEW

File: Source/Libraries/CorruptCore/Objects.cs [hunk 1]
```diff
@@ -1723,6 +1663,52 @@ public List<BlastUnit> GetRasterizedUnits(string vmdToRasterize = null)
             return l.Layer;
         }
 
+        private bool SetDomainAndAddress(bool useSourceDomain)
```
Comment (NarryG) [inline Source/Libraries/CorruptCore/Objects.cs:1666]: Could you rename this to something slightly more descriptive? Maybe SetRealDomainAndAddress or something. The idea behind the code here is it converts a VMD to the underlying domains, but it has this recursive breakdown element to it where if the VMD is non-contiguous, it needs to recursively break the unit down. Might make more sense to do a bigger refactor at some point to make that code flow more clear.

Comment (scowalt) [reply Source/Libraries/CorruptCore/Objects.cs]: Changed to `SetRealDomainAndAddress`.

Thanks for the context. Sounds like this code may be worth revisiting in the future.

**Comment by NarryG** [comment]
Here come the conflicts.

**Comment by scowalt** [comment]
Merge conflicts resolved

**Pull request merged**